llvm/lib/Analysis/MemoryBuiltins.cpp

```diff
@@ -1138,8 +1138,8 @@ SizeOffsetValue ObjectSizeOffsetEvaluator::visitAllocaInst(AllocaInst &I) {
   if (!I.getAllocatedType()->isSized())
     return ObjectSizeOffsetEvaluator::unknown();
 
-  // must be a VLA
-  assert(I.isArrayAllocation());
+  // must be a VLA or vscale.
+  assert(I.isArrayAllocation() || I.getAllocatedType()->isScalableTy());
```

Comment:
Does this mean we were violating this assertion for vscale allocas before this change?

Reply:
Yes, DEBUG crashes here, NDEBUG crashes casting `DL.getTypeAllocSize` to const